django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
```

Comment:
I think the session cookie should be independently configurable. Technically there's no reason we'd need two settings, it would be possible to just have `CSRF_SESSION_KEY`, and use it both as the session key and the trigger to use sessions at all (if it's not set, we don't). Not sure if that's actually better than just having two settings.